app/controllers/users_controller.rb

```diff
@@ -1740,6 +1740,61 @@ def bookmarks
     end
   end
 
+  USER_MENU_BOOKMARKS_LIST_LIMIT = 20
+  def user_menu_bookmarks
+    user = fetch_user_from_params
+    guardian.ensure_can_see_notifications!(user)
```

Comment:
Since this is only used for the user menu, is there a need for us to allow admin to view it? I think we can just let the current user view their own notifications and no one else.

Reply:
The path for this action `/u/:username/user-menu-bookmarks`, should I change it so it doesn't have a dynamic `:username` segment?

Reply:
Yea I think we can just change it. No point exposing it to admin when we don't need to yet.

Reply:
Ok, I'm going to change it to `/bookmarks/user-menu`, any objections/feedback on the path?

Reply:
Nope none from me

Reply:
O wait but you're not just fetching bookmarks but notifications too. I'll keep it in the users controller in this cas.e

Reply:
so same path (`/bookmarks/user-menu`) but it goes to the users controller? maybe a top level path `/user-menu-bookmarks`?

Reply:
/u/user-menu-bookmarks perhaps?

Reply:
haha I don't want to think what happens if someone signs up with a `user-menu-bookmarks` username 🙈

Reply:
O that is annoying... Just keep it as `/u/:username/user-menu-bookmarks` then? Just that this path is only for the current user.

Reply:
> haha I don't want to think what happens if someone signs up with a user-menu-bookmarks username

Hmm actually we have this problem with the other routes too. You can break certain features of Discourse by signing up with usernames that matches existing routes.

Reply:
> Hmm actually we have this problem with the other routes too. You can break certain features of Discourse by signing up with usernames that matches existing routes.

Yeah you're right... I hope Rails prefers the route with a static segment over a route with a dynamic segment when there are 2 routes matching a request's path?

Reply:
It is based on the order, so you have to declare it before the dynamic routes.

Reply:
Ok having thought about this some more, I don't think we should go with `/u/user-menu-bookmarks` because if someone signs up with that username, I think they'll get a broken profile page (since the static route is declared first) and if they cause trouble, site mods will have a hard time banning them because their profile is broken.

I don't know, I feel like we should just leave it as-is?

Reply:
Yea `/u/:username/user-menu-bookmarks` is fine IMO as long as we just restrict it to the current user.

Reply:
so we raise an error if `param[:username] != current_user.username`?

Reply:
Yup